src/v/kafka/server/group_manager.cc

```diff
@@ -615,6 +623,62 @@ ss::future<> group_manager::reload_groups() {
     co_await ss::when_all_succeed(futures.begin(), futures.end());
 }
 
+ss::future<group_offsets_snapshot_result>
+group_manager::snapshot_groups(const model::ntp& ntp) {
```

Comment:
this function looks good to me. i'm curious if we have something like it already, since presumably we are check pointing offset state into the underlying partition already?

Reply:
@rystsov @bharathv for looking at the locking here, especially the catchup_lock.

Reply:
`catchup_lock` looks fine; when we load the data from the partition we take write lock so read lock guarantees that the loaded data is complete

Reply:
Thank you!